### PR TITLE
[DOC-8971] Update docs for renamed shutdown settings

### LINKED ISSUE #18684: Feedback: What's New in v23.2
Page: https://cockroachlabs.com/docs/releases/v23.2.html

## What is the reason for your feedback?

[ ] Missing the information I need

[ ] Too complicated

[ ] Out of date

[ ] Something is broken

[x] Other

## Additional details
The connection.timeout and transaction_timeout alias aren’t correct, comparing w/ cluster setting page.
Can someone review and correct it please?

### PR DESCRIPTION
**Pull request by mdlinville**
Fixes #18684
Fixes [DOC-8971]

Now that Cluster Settings pages show both the old and new names and the deprecation message has been softened, I can update the docs to use the new names.

- Also correct setting name errors in old release notes
- Also add details about key shutdown settings to the Cloud Production Checklist, per Carlo

Previews:
[releases/v23.2.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/releases/v23.2.html)
[src/current/v23.2/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v23.2/monitoring-and-alerting.html)
[src/current/v23.2/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v23.2/node-shutdown.html)
[src/current/v24.1/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.1/monitoring-and-alerting.html)
[src/current/v24.1/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.1/node-shutdown.html)
[src/current/v24.2/monitoring-and-alerting.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.2/monitoring-and-alerting.html)
[src/current/v24.2/node-shutdown.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/v24.2/node-shutdown.html)
[src/current/cockroachcloud/production-checklist.md](https://deploy-preview-18689--cockroachdb-docs.netlify.app/docs/cockroachcloud/production-checklist.html)

### CODE REVIEW

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [inline src/current/cockroachcloud/production-checklist.md:71]: @carloruiz PTAL

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md
Comment (rmloveland) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: I do not understand this. Were the previous release notes incorrect?  

If we go back and say this about the v23.2.0-alpha.3 either the docs were wrong (totally possible) or ... ?

File: src/current/cockroachcloud/production-checklist.md
Comment (rmloveland) [inline src/current/cockroachcloud/production-checklist.md:69]: this section LGTM from a docs editing POV but I defer to @carloruiz for technical review

**Review by rmloveland — APPROVED** [review]
LGTM from Docs POV modulo any technical review comments that may need addressing 

EDIT: oh and I need to understand why we are back changing those old alpha release notes, that doesn't make sense to me unless those docs were actually wrong

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md
Comment (mdlinville) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: The release note was incorrect. This fixes it.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:75]: I'd prefer not to mention this as it is not actionable for the customer and makes the doc harder to digest.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:81]: I'd also not mention this as it's not actionable for customers and sounds scary.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:69]: I'd move this section below `Keeping connection current` since the desired value for `connection_wait` should be set equal to or above the max lifetimes set on their connections.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:79]: The hard limit for termination should not be communicated externally as we (the infra team) want to be flexible with this value. We should distill the termination info through the cluster settings only to avoid unnecessary complexity on the customer side and avoid customer asks that we can't accommodate.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:76]: I'd reword this a bit: "how long a crdb node waits for client SQL connections to close before forcibly closing the connection". 

Below, I'd also note the tradeoff involved when setting this value. Increasing this value will result in slower cluster operations (version upgrades, decommissioning nodes, etc). Each node shutdown will now wait `server.shutdown.connections.timeout` as part of the shutdown process. But increasing it may be required to avoid connection interruptions during these maintenance ops.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:77]: We should note that 90s is the highest supported value. Increasing it beyond this value can result in tx failures during cluster operations.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:69]: On further read, I think this section should be folded under "Keeping connections current"

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md
Comment (rmloveland) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.3.md:108]: Thank you for fixing it!

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done, created a new Networking considerations heading and moved Keeping Connections Current (merged with the shutdown settings section) and Network Authorization into it.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:86]: Let's drop this line. As a developer it is easier to digest an upper bound on a single variable than an upper bound on an equation. Let's give upper bounds on each variable separately.  

Also, this line and everything below do not apply to Serverless.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:90]: Can we add a column for max value?

**Review by carloruiz — COMMENTED** [review]
Last few comments are about splitting dedicated and serverless recommendations. For serverless it is simpler, we just tell them to configure their connection lifetimes to a max of 30m. The cluster setting considerations only apply to dedicated.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md]: Done. If I understand it correctly, we default to the max for `server.shutdown.transactions.timeout` so I changed the last sentence to "Decreasing this value ...."

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:95]: `s/rolled back/cancelled`. 

Add: this value should be set to the longest expected transaction. Transactions lasting longer than `server.shutdown.transactions.timeout` can be interrupted during node restarts.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:94]: Add: connections with lifetimes over 30m can be interrupted during node restarts.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:85]: Add: set the connection lifetime in the connection pool to 30m.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:81]: Let's merge Keeping Connections Current and SQL Connection Handling.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:88]: After this line `In your application server, set the maximum lifetime of a connection to between 5 and 30 minutes.` let's also tell them to set `server.shutdown.connections.timeout` equal to the max lifetime on their connections.

**Review by carloruiz — COMMENTED** [review]
Let's have a quick call tomorrow to hammer this out with faster iteration cycles.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:94]: Done with some tweaks. PTAL.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:81]: Done with some wording adjustments.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:88]: Done

**Comment by mdlinville** [comment]
> Let's have a quick call tomorrow to hammer this out with faster iteration cycles.

Sorry I wasn't able to do this today but if you have a lot more feedback on this round, let's do that tomorrow or Thursday.

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:85]: `s/reset after 30 minutes/reset after server.shutdown.connections.timeout minutes`

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:91]: typo, 3 minutes -> 30 minutes

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [inline src/current/cockroachcloud/production-checklist.md:82]: Serverless customers should also configure their connection pools with connections with max lifetimes of 30m

File: src/current/cockroachcloud/production-checklist.md
Comment (carloruiz) [reply src/current/cockroachcloud/production-checklist.md:74]: I'd restructure this a little bit. The top level title could be "SQL connection Handling" with a subsection on "connection pooling" since this is performance related, then a section for "keeping connections current" since this is for reliability.

**Review by carloruiz — COMMENTED** [review]
OK I think we're 95% there

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:85]: Done

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:74]: Apologies, I misinterpreted what you meant by "merge". Fixed.

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [reply src/current/cockroachcloud/production-checklist.md:82]: Done

File: src/current/cockroachcloud/production-checklist.md
Comment (mdlinville) [inline src/current/cockroachcloud/production-checklist.md:90]:
```suggestion
[`server.shutdown.connections.timeout`]({% link {{ site.current_cloud_version }}/cluster-settings.md %}#setting-server-shutdown-connection-wait)<br />Alias: `server.shutdown.connection_wait` | 0 seconds  | 30 minutes (1800 seconds) | How long to wait for client connections to drain before forcibly disconnecting them from the node. A connection with a lifetime that exceeds `server.shutdown.connections.timeout` may be interrupted during node restarts.
```

**Comment by mdlinville** [comment]
I've resolved all feedback so I'll go ahead and merge, and we can address any feedback as a fast follow.

**Pull request merged**
